### PR TITLE
Feature/ab#147 registration overview

### PR DESCRIPTION
**Pull request by vyruz1986**

### CODE REVIEW

File: Singer.API/ClientApp/src/app/modules/core/components/dailybasis-registrations/dailybasis-registrations.component.html [hunk 1]
```diff
@@ -17,20 +17,20 @@
          </th>
          <td mat-cell *matCellDef="let row" class="user-column">
             <mat-icon
-               *ngIf="getSlotRegistrationStatus(row.registrations, user.careUserId) == 1"
+               *ngIf="getSlotRegistrationStatus(row.registrations, user.careUserId) === RegistrationStatus.Pending"
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/core/components/dailybasis-registrations/dailybasis-registrations.component.html:20]: Good call on using the enums

File: Singer.API/ClientApp/src/app/modules/core/core-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Routes, RouterModule } from '@angular/router';
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/core/core-routing.module.ts:1]: Didn't expect the CoreModule to have a routing

File: Singer.API/ClientApp/src/app/modules/dashboard/components/registration-overview/registration-overview.component.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+         <mat-datepicker #dateFrom></mat-datepicker>
+      </mat-form-field>
+      <mat-form-field>
+         <input matInput #searchDateTo [matDatepicker]="dateTo" placeholder="Datum tot" (dateChange)="loadData()" />
+         <mat-datepicker-toggle matSuffix [for]="dateTo"></mat-datepicker-toggle>
+         <mat-datepicker #dateTo></mat-datepicker>
+      </mat-form-field>
+      <mat-form-field>
+         <mat-select mat-input placeholder="Type Registratie" #searchType (selectionChange)="loadData()" multiple>
+            <mat-option *ngFor="let type of registrationTypes | keys" [value]="type.key">
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/dashboard/components/registration-overview/registration-overview.component.html:20]: Nice!

File: Singer.API/ClientApp/src/app/modules/dashboard/components/registration-overview/registration-overview.component.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      super.loadData();
+   }
+
+   onCareUserFilterChange(careUsers: CareUser[]) {
+      this.searchDTO.careUserIds = careUsers.map(x => x.userId);
+      this.loadData();
+   }
+
+   changeRegistration(registrationStatus: RegistrationStatus, registration: RegistrationOverview) {
+      console.log(registration);
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/dashboard/components/registration-overview/registration-overview.component.ts]: `console.log`
Also, isn't this logic duplicated?

File: Singer.API/ClientApp/src/app/modules/shared/components/registration-status-toggle/registration-status-toggle.component.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<mat-button-toggle-group class="status-icon" [value]="registrationStatus" (change)="changeStatus($event)">
+   <mat-button-toggle [value]="RegistrationStatus.Accepted" matTooltip="Inschrijving goedkeuren">
+      <mat-icon class="registration-accepted">check_circle_outline</mat-icon>
+   </mat-button-toggle>
+   <mat-button-toggle
+      [disabled]="'true'"
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/shared/components/registration-status-toggle/registration-status-toggle.component.html]: Could be `disabled=true` (or even just `disabled`)

File: Singer.API/ClientApp/src/app/modules/shared/components/registration-status/registration-status.component.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+   selector: 'app-registration-status',
+   templateUrl: './registration-status.component.html',
+   styleUrls: ['./registration-status.component.css'],
+})
+export class RegistrationStatusComponent implements OnInit {
+   @Input('status') registrationStatus: RegistrationStatus;
+   RegistrationStatus = RegistrationStatus;
+   constructor() {}
+
+   ngOnInit() {}
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/shared/components/registration-status/registration-status.component.ts]: ngOnInit is not needed.

File: Singer.API/Controllers/EventController.cs [hunk 7]
```diff
@@ -347,19 +347,18 @@ public async Task<ActionResult<EventRegisterDetailsDTO>> GetEventRegisterDetails
             StartDate = singerEvent.StartDateTime,
             Title = singerEvent.Title
          };
-         List<EventRelevantCareUserDTO> careUsers;
-         if (!User.IsInRole(Roles.ROLE_ADMINISTRATOR))
+         List<EventRelevantCareUserDTO> careUsers = new List<EventRelevantCareUserDTO>();
```
Comment (BerendWouters) [inline Singer.API/Controllers/EventController.cs]: assignment can be `var`

File: Singer.API/Models/Registration.cs [hunk 8]
```diff
@@ -9,32 +9,32 @@ public class Registration : IIdentifiable
...
 
       [ForeignKey(nameof(DaycareLocation))]
       public Guid? DaycareLocationId { get; set; }
-      public EventLocation DaycareLocation { get; set; }
+      public virtual EventLocation DaycareLocation { get; set; }
 
       public DateTime StartDateTime { get; set; }
       public DateTime EndDateTime { get; set; }
-      private Registration()
+      public Registration()
```
Comment (BerendWouters) [inline Singer.API/Models/Registration.cs]: Why is this suddenly public?

**Review by BerendWouters — APPROVED** [review]

File: Singer.API/ClientApp/src/app/modules/dashboard/components/registration-overview/registration-overview.component.ts (hunk 4 above)
Comment (vyruz1986) [reply Singer.API/ClientApp/src/app/modules/dashboard/components/registration-overview/registration-overview.component.ts]: Removed console.log.
Same functions are indeed in registration status change in event component, but I didn't want to generalize that code yet. I'll add a TODO

**Pull request merged**